### ISSUE TITLE
Resource tests not running properly

### ISSUE THREAD

**Issue by typerandom** [stormpath/stormpath-sdk-node#334]
When running `npm tests` not all `sp.resource.*` tests run properly.

**Comment by robertjd** [comment]
This issue has broaded in scope.  The following work should also be done as part of this issue:

* Remove the Gruntfile, and run mocha tests directly
* The tests in `sp.resource.collectionResource_test.js` are a real mess, and use nock which is brittle.  These tests should be re-written without nock.

**Comment by timothyej** [comment]
I have a pretty good idea of what the problem is. In many of the tests, there are assignments inside the `describe` blocks, which is not very good if they're used in deeper `describe` blocks. For example, in `sp.resource.group_test.js`:

```javascript
describe('Group resource', function () {
  var dataStore = new DataStore({client: {apiKey: {id: 1, secret: 2}}});

  describe('add account', function () {
    describe('if group href not set', function () {
      var group = new Group(dataStore);
      ...
    });
    ...
  });
  ...
});
```

Instead, the assignments should be done in `before`/`beforeEach` blocks while the declarations are in the `describe` blocks:

```javascript
describe('Group resource', function () {
  var dataStore;

  beforeEach(function () {
    dataStore = new DataStore({client: {apiKey: {id: 1, secret: 2}}});
  });

  describe('add account', function () {
    describe('if group href not set', function () {
      var group;

      beforeEach(function () {
        group = new Group(dataStore);
      });
      ...
    });
    ...
  });
  ...
});
```

You can read more about it here: [Variable in outer describe block is undefined when accessing in inner describe block with Mocha test](http://stackoverflow.com/questions/21470349/variable-in-outer-describe-block-is-undefined-when-accessing-in-inner-describe-b)

**Comment by typerandom** [comment]
@timothyej I agree with your refactoring. But unfortunately it's not such a simple fix for this issue.

**Comment by typerandom** [comment]
`sp.authc.apiKey.js` is also not running.

```
stormpath-sdk-node git:(fix-silently-failing-tests) mocha test/sp.authc.apiKey.js --no-timeouts


  authc
    ApiKey class
      1) should expose id and secret as fields
      2) should hide secret


  0 passing (10ms)
  2 failing

  1) authc ApiKey class should expose id and secret as fields:
     TypeError: Cannot read property 'be' of undefined
      at Context.<anonymous> (test/sp.authc.apiKey.js:14:23)

  2) authc ApiKey class should hide secret:
     TypeError: Cannot read property 'contain' of undefined
      at Context.<anonymous> (test/sp.authc.apiKey.js:18:31)
```

**Comment by timothyej** [comment]
This fails because `chai` hasn't been loaded in this file. Adding these two lines to the top makes the tests run:

```javascript
var chai = require('chai');
chai.should();
```

But this is handled by Grunt. And the reason why Grunt isn't running this test is because it doesn't end with `_test.js`. So it isn't *failing silently*, it's just not running at all ;)

**Comment by typerandom** [comment]
Nice find. I haven't had time to dig deeper into that one. The "failing silently" part was meant for the `sp.resource.*` tests. This is just a test that I stumbled upon that also weren't being run.

**Comment by timothyej** [comment]
I've addressed this issue in this PR: #354

**Comment by typerandom** [comment]
:+1:
